source/daplink/interface/main.c

```diff
@@ -107,30 +107,17 @@ static main_led_state_t msc_led_state = MAIN_LED_FLASH;
 main_usb_connect_t usb_state;
 static bool usb_test_mode = false;
 
-static U64 stk_timer_30_task[TIMER_TASK_30_STACK / sizeof(U64)];
-static U64 stk_main_task[MAIN_TASK_STACK / sizeof(U64)];
-
-// Timer task, set flags every 30mS and 90mS
-__task void timer_task_30mS(void)
+// Timer task, set flags every 30mS
+void timer_task_30mS(void)
 {
-    uint8_t i = 0;
-    os_itv_set(3); // 30mS
-
-    while (1) {
-        os_itv_wait();
-        os_evt_set(FLAGS_MAIN_30MS, main_task_id);
-
-        if (!(i++ % 3)) {
```

Comment:
You could leave the logic in `timer_task_30mS()` for setting the 90 ms event (for both interface and bootloader). Just change `i` to be static. This would prevent you from having to move around the chunks of code in the main task.